### ISSUE TITLE
X-sheet: Choice of left-to-right/right-to-left Column order view

### ISSUE THREAD

**Issue by callybug** [opentoonz/opentoonz#5099]
### Is your feature request related to a problem? Please describe

On a traditional exposure sheet, the bottom-most level (usually background) is on the right, with each additional level being left of the level beneath it:
Overlay, Top-most animation cel, Middle animation cel, Bottom animation cel, Background artwork, (then Camera/pegbar movement instructions).
In OpenToonz, this is the opposite.

I can see how going left-to-right makes more intuitive sense to animators familiar with the Flash/timeline format, but going back and forth between paper and computer, it's something I have to relearn each time.

### Describe the solution you'd like

Is it possible to add an option to view the X-Sheet going top-most layer left, bottom layer right? Thank you.

**Comment by RodneyBaker** [comment]
I hadn't considered this before for the xsheet but we did have a discussion about that order when the Timeline was created.
Would love to see this option to invert the order in the Timeline as well.
In general the idea being considered here is whether the layer is closer to or farther away from the camera.

This request makes good sense to me in that:
The order of exposure can seem backward in that a layer/column closest to the camera is under all the other layers when logically one would think a layer closer to the camera would be on top. 
Having said this my sense is that the default should be maintained as is in that as we draw the norm is to draw something first, then draw something new and the newest is automatically placed on top of the old. (with obvious ability to invert these manually as needed).

You've got me thinking a bit more about the logic behind the order presented.

**Comment by callybug** [comment]
Hi @RodneyBaker !
The way you explain the logic does make sense, yet OpenToonz's Camera column is on the far-left on the X-sheet (left of the lowest layer; a mirror image of a "traditional" exposure sheet). In the horizontal Timeline view, the top layer being highest up and "closest" to the Camera does make sense to me – but I imagine it'd get in the way less staying underneath all the animation layers, where the most work goes on.

>  I'd love to know more about the system you are using @callybug that reverses this in the xsheet.

The clearest example I can find is in the Animator's Survival Kit book, where it visualises the cels on top each other, with arrows pointing going right-to-left, BG, 1, 2, 3...
![richardwilliamsxsheet](https://github.com/opentoonz/opentoonz/assets/113436322/1b719308-9c99-4e2f-a590-e7b82f25ce81)
There's images online of various studios' exposure sheets throughout history. Although – now you got me searching – I did find a couple studios' exposure-sheets going bottom-to-top left-to-right, with the camera movements right of the top-most cel.

The reason I think the camera instructions were typically by the _lowest_ level is because the cameraman will have an easier time following the sheet if the camera instructions were closest to the most populated level (there's always at least one level of animation, even just a background). And the Action and Dialogue columns for the animator are usually on the left; so the animator and cameraman have their respective "sides to work". The animator doesn't need the technical camera instructions on his "side", which is why I think OpenToonz has the Camera where it is at the moment. It's just mirrored.*

If there's also been discussion regarding where the Camera should go in the Timeline view (I've only used the X-sheet view, so I didn't know) maybe there could be a menu in Preferences to choose the bottom-to-top presentation, and also Camera column position defaults, for both Timeline and X-sheet view (or have the camera freely draggable).
![xsheetmenuidea](https://github.com/opentoonz/opentoonz/assets/113436322/33a67798-b784-493f-87d0-dee2c325abfa)
I realise there's a quickly rising risk of falling into the "FOSS = choice" fallacy with this, my apologies. 


*(possibly to make it easier for the scanners/compositors, since the focus wasn't paperless animation but digital compositing? I'm only assuming that's what most of Toonz's life has been up until recently.)

**Comment by RodneyBaker** [comment]
Thanks @callybug for the detailed response.
Many classic xsheets do indeed reverse the entire listing; moving even the camera column all the way to the right.
You've got me interested in studying xsheets all over again.

An issue we'll have to account for is how when the order changes the camera column moves as well and that is where I think the difficulty might appear as I'm not sure anyone will want the camea column to be all the way to the right in the xsheet... or might they?  The key being that column number 1 is relatively the closest to the camera column regardless of which direction of flow the columns go.

I like your drawing except for one thing; a level is not a direct equivalent for a column.
We can multiple levels... even of different level types... inside a column.
As such reversing level order means something entirely different to me.
I do use the terms 'layer' and 'column' interchangably because columns represent content (levels) that are layered over one another.

You are on to something here it's just not entirely clear yet what that is.

**Comment by callybug** [comment]
> I like your drawing except for one thing; a level is not a direct equivalent for a column.

Oh! Sorry, yes @RodneyBaker you're exactly right! _Column_ order is what I've been meaning to address (I'll go back and make edits in a minute).

Re: the Camera column being on the right. The columns would go right-to-left.
A mirror image of the X-Sheet in OpenToonz would be close to how a traditional exposure sheet is.

Did a mockup then realised something else, and also probably why the default is the way it is.
![xsheet idea](https://github.com/opentoonz/opentoonz/assets/113436322/3a8a4c3f-3f69-482d-9fd2-d4b79be118ba)
The frame numbers are often written in the camera column near the rightmost column to compare with, and reference for drawing numbers (otherwise it has its own column – sometimes in conjunction with the Dial. column):
![Screenshot_2023-09-17_17-53-28](https://github.com/opentoonz/opentoonz/assets/113436322/dae8b0cf-62cc-401a-baba-19663f190fa2)



I think it's the other way around in OpenToonz so the frame number only has to be displayed once, and near the Camera column on the left-side. For the UI's present design, the devs made the right choice – but maybe there's some other animators who find themselves having to think twice throughout animating a scene.
Tried another:
![xsheetmockup2](https://github.com/opentoonz/opentoonz/assets/113436322/d6877f97-0b04-429b-8702-0dc47fde5dc1)
The camera column can at least be hidden to get around that being in the wrong place. 
Apologies, this has got more complicated than I first realised. (I might try changing the column order on my exposure sheets when I next animate on paper; maybe I'll adapt that way!)

**Comment by RodneyBaker** [comment]
Yes, you've demonstrated that this is indeed a little more complicated than it seems upon first look.
And... you've reminded me of how forgetful we can be... I had convinced myself that paper-based xsheets were numbered left to right (after camera column) and no... I pull out some of my paper based xsheets from long ago and that isn't the case at all.  They are closer to what you describe originally.

Interestingly, If we use File > Export Xsheet as PDF... the column headers aren't even numbered.
Instead the Column title is place there and in the columns below the level name is exposed.
I think I like this number-less approach although I can see where it might be useful to know and use the numbers.

Someone just needs to fix this mess and be done with it.  
If we can build a time machine I think we can go back and fix it for good!  ;)

Aside:  I can imagine that for a left handed person having the numbers on the right might be useful.
For me having that order reversed looks really strange.
